shared/src/commonMain/kotlin/org/jetbrains/kotlinconf/screens/SettingsScreen.kt

```diff
@@ -0,0 +1,244 @@
+
+            val model = notiState.model
+            LaunchedEffect(model) {
+                onNotificationSettingsChange(model)
+            }
+        }
+    }
+}
+
+private val themes = listOf(Theme.SYSTEM, Theme.DARK, Theme.LIGHT)
+
+@Composable
+private fun ThemeSelector(
+    currentTheme: Theme,
+    onThemeChange: (Theme) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Row(
+        horizontalArrangement = Arrangement.spacedBy(8.dp),
+        modifier = modifier
+    ) {
+        themes.forEach { theme ->
+            ThemeBox(
+                theme = theme,
+                isSelected = currentTheme == theme,
+                onClick = { onThemeChange(theme) },
+                modifier = Modifier.weight(1f),
+            )
+        }
+    }
+}
+
+@Composable
+private fun ThemeBox(
+    theme: Theme,
+    isSelected: Boolean,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = modifier
+            .clickable(
+                onClick = onClick,
+                indication = null,
+                interactionSource = remember { MutableInteractionSource() },
+            )
+            .semantics(mergeDescendants = true) {},
+    ) {
+        Box(
+            contentAlignment = Alignment.Center,
+            modifier = Modifier
+                .then(
+                    if (isSelected) {
+                        Modifier
+                            .border(
+                                width = 2.dp,
+                                color = KotlinConfTheme.colors.primaryBackground,
+                                shape = RoundedCornerShape(12.dp)
+                            )
+                    } else Modifier
+                )
+                .padding(6.dp)
+                .border(
+                    width = 2.dp,
+                    color = KotlinConfTheme.colors.strokePale,
+                    shape = RoundedCornerShape(8.dp)
+                )
+                .clip(RoundedCornerShape(8.dp))
+                .heightIn(max = 112.dp)
+                .aspectRatio(1f)
+        ) {
+            Image(
+                painter = painterResource(
+                    when (theme) {
+                        Theme.SYSTEM -> Res.drawable.theme_system
+                        Theme.LIGHT -> Res.drawable.theme_light
+                        Theme.DARK -> Res.drawable.theme_dark
+                    }
+                ),
+                contentDescription = null,
+                modifier = Modifier.fillMaxSize()
+            )
+        }
+        Spacer(Modifier.height(8.dp))
+        StyledText(
+            text = stringResource(
+                when (theme) {
+                    Theme.SYSTEM -> Res.string.settings_theme_system
+                    Theme.LIGHT -> Res.string.settings_theme_light
+                    Theme.DARK -> Res.string.settings_theme_dark
+                }
+            ),
```

Comment:
I'd provide respective resources as parameters directly to this Composable.
It would increase the number of parameters just by one but will be more straightforward.
At a call site, replacing loop with three calls wouldn't make the code any worse as well.

Reply:
For me it's cleaner if the `ThemeBox` can be used with just the `Theme` value passed in as a parameter, I'll leave it as-is for now.